### PR TITLE
Handle Signer Rejection

### LINKED ISSUE #3153: When signer rejects transaction, transaction modal should reflect the state
Currently when sending a transaction, rejecting it in the signer pops up an error (expected), however the "waiting for hash" still sits spinning. This should handle the error as well. 

Same goes for when sending in the contract interface.

### PR DESCRIPTION
**Pull request by ngotchac**
Fixes #3153

This also introduces real Errors with the code and type from Core

### CODE REVIEW

File: js/src/api/transport/ws/ws.js [hunk 1]
```diff
@@ -18,6 +18,7 @@ import { keccak_256 } from 'js-sha3'; // eslint-disable-line camelcase
 
 import { Logging } from '../../subscriptions';
 import JsonRpcBase from '../jsonRpcBase';
+import TransportError from '../error';
```
Comment (jacogr) [inline js/src/api/transport/ws/ws.js]: Any reason why this is only introduced to the WS transport and not the HTTP transport (as used by dapps) as well?

Comment (ngotchac) [inline js/src/api/transport/ws/ws.js]: It is :)

Comment (jacogr) [inline js/src/api/transport/ws/ws.js]: I swear... I'm going blind.

**Comment by jacogr** [comment]
Played, looking good.

**Pull request merged**